src/utils/getFilteredEntityFields.ts

```diff
@@ -0,0 +1,172 @@
+};
+
+/**
+ * Returns a list of {@link EntityFieldNameAndSchema} given a {@link YextSchemaField}.
+ * @param schemaField The schema field for which to get the field names.
+ * @returns Entity field names and their corresponding schema fields.
+ */
+const getEntityFieldNames = (
+  schemaField: YextSchemaField
+): EntityFieldNameAndSchema[] => {
+  const entityFieldNames: EntityFieldNameAndSchema[] = [];
+  if (DEFAULT_DISALLOWED_ENTITY_FIELDS.includes(schemaField.name)) {
+    return entityFieldNames;
+  }
+
+  return walkSubfields(schemaField, "");
+};
+
+const walkSubfields = (
+  schemaField: YextSchemaField,
+  fieldName: string
+): EntityFieldNameAndSchema[] => {
+  const fieldNameInternal = fieldName
+    ? `${fieldName}.${schemaField.name}`
+    : schemaField.name;
+
+  if (
+    TOP_LEVEL_ONLY_FIELD_TYPES.includes(schemaField.definition.typeName) ||
+    TOP_LEVEL_ONLY_FIELD_TYPES.includes(schemaField.definition.typeRegistryId)
+  ) {
+    return [{ name: fieldNameInternal, schemaField: schemaField }];
+  }
+
+  if (
+    schemaField.children?.fields &&
+    schemaField.children?.fields?.length > 0
+  ) {
+    const fieldNames: EntityFieldNameAndSchema[] = [];
+    for (const child of schemaField.children.fields) {
+      fieldNames.push(...walkSubfields(child, fieldNameInternal));
+    }
+    return fieldNames;
+  }
+
+  return [{ name: fieldNameInternal, schemaField: schemaField }];
+};
+
+const getTypeFromSchemaField = (schemaField: YextSchemaField) => {
+  return (
+    schemaField.definition.typeName ||
+    schemaField.definition.typeRegistryId ||
+    Object.entries(schemaField.definition.type)[0][1]
+  );
+};
+
+const getEntityTypeToFieldNames = (
+  schemaFields: YextSchemaField[],
+  filter: {
+    includeSubfields: boolean;
+  }
+): Map<string, string[]> => {
+  return schemaFields.reduce((prev: Map<string, string[]>, field) => {
+    if (field.definition.isList) {
+      return prev;
+    }
+
+    if (filter.includeSubfields) {
+      const fieldNameToSchemas = getEntityFieldNames(field); // already recursed
+      if (fieldNameToSchemas.length === 0) {
+        return prev;
+      }
+      for (const fieldNameToSchema of fieldNameToSchemas) {
+        const typeName = getTypeFromSchemaField(fieldNameToSchema.schemaField);
+        if (!typeName) {
+          continue;
+        }
+
+        if (!prev.has(typeName)) {
+          prev = prev.set(typeName, [fieldNameToSchema.name]);
+        } else {
+          prev.get(typeName)?.push(fieldNameToSchema.name);
+        }
+      }
+    } else {
+      const typeName = getTypeFromSchemaField(field);
+      if (!typeName) {
+        return prev;
+      }
+
+      if (!prev.has(typeName)) {
+        prev = prev.set(typeName, [field.definition.name]);
+      } else {
+        prev.get(typeName)?.push(field.definition.name);
+      }
+    }
+
+    return prev;
+  }, new Map<string, string[]>());
+};
+
+export const getFilteredEntityFields = (filter?: RenderEntityFieldFilter) => {
```

Comment:
Move the filtering stuff into getEntityTypeToFieldNames directly and remove this function. It was my first pass version that getEntityTypeToFieldNames supersedes now and I didn't remove it from my PR.

Reply:
Talked offline, we just want to remove the `.filter((field) => field.children === undefined)` filter on line 141.